modules/apps/frontend-js/frontend-js-bundle-config-extender/src/main/java/com/liferay/frontend/js/bundle/config/extender/internal/JSBundleConfigRegistry.java

```diff
@@ -126,31 +122,28 @@ protected void activate(
 			ComponentContext componentContext, Map<String, Object> properties)
 		throws Exception {
 
-		if (_serviceTracker != null) {
-			_serviceTracker.close();
+		if (_serviceTrackerMap != null) {
+			_serviceTrackerMap.close();
 		}
 
 		_bundleContext = componentContext.getBundleContext();
 
-		_serviceTracker = ServiceTrackerFactory.open(
-			_bundleContext,
-			"(&(objectClass=" + ServletContext.class.getName() +
-				")(osgi.web.contextpath=*))",
-			this);
+		_serviceTrackerMap = ServiceTrackerMapFactory.openSingleValueMap(
+			_bundleContext, ServletContext.class, "osgi.web.contextpath", this);
 	}
 
 	@Deactivate
 	protected void deactivate() {
-		_serviceTracker.close();
+		_serviceTrackerMap.close();
 
-		_serviceTracker = null;
+		_serviceTrackerMap = null;
 	}
 
 	private BundleContext _bundleContext;
 	private final Map<ServiceReference<ServletContext>, JSConfig> _jsConfigs =
```

Comment:
@MrJohn1911 this looks really odd, can you just make this "_jsConfigs" the service tracker map and avoid adding the EagerServiceTrackerCustomizer?
The service tracking logic does not seem to have any side affects, there should be no reason to be eager.